Issue#45 refactoring script and test

hrsano645 opened the pull request:
<!-- Issue 番号がない PR は受け付けません。 -->
<!-- We don't accept PRs which has no Issue ID. -->

## 👏 解決する issue / Resolved Issues
- close #45

## 📝 関連する issue / Related Issues
- #43 
- #42 

## ⛏ 変更内容 / Details of Changes
<!-- 変更を端的に箇条書きで -->
<!-- List down your changes concisely -->
- 大規模なリファクタリングを行いました。
- オープンデータの入力とdata.jsonのそれぞれのデータの生成部分を関数にして分離しました
- それに伴い関数のテストを追加しました
- リファクタリングなので、生成結果は以前の状態と変わりません。
  - https://github.com/hrsano645/covid19-gen-datajson-shizuokapref/tree/dad95af99af58fd1d8d997e0b8daeb7129a93889 のcommitでデータの生成をテストして同様の結果となることを確認済みです。

## 📸 スクリーンショット / Screenshots
<!-- 参考画像があれば添付してください -->

hrsano645 committed:
Squashed commit of the following:

commit 18814adb73a19eb014fc65f3ac7a0e97bb49b2de
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Thu Feb 25 21:31:43 2021 +0900

    change json template into gen_funcs

commit bc112c86d2b1eae3690af892f5e94652752f787e
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Thu Feb 25 21:18:08 2021 +0900

    日付関係の処理やコメントを整理

commit d9deb4d1b7f9433b289e7308605ebcfdfdea8bb6
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Thu Feb 25 21:15:44 2021 +0900

    change datetime formatted str func

commit c4ff785c4bfc3cb9a5edd48e1ad394f1b8f2649e
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Thu Feb 25 21:14:17 2021 +0900

    update testcase and data

commit 6c0329d2faade0554ff4378dce52fbbe7d2e14f2
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Thu Feb 25 20:28:56 2021 +0900

    fix 新しいオープンデータへの対応

commit 861c5498db30279d6eff87182ed8ce274429bea6
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Thu Feb 25 20:13:21 2021 +0900

    fix 新しいオープンデータの対応

commit 4b4e31da220a80feaadf8fc1c845c24c2449f131
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Thu Feb 25 20:11:41 2021 +0900

    remove comment

commit 727207be79cf944f4c32b66808bfa9951171fbde
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Thu Feb 25 20:09:55 2021 +0900

    add new opendata
    ref #43

commit 7653407ff6a61dc1c0cdb46aef074e4b8f40f450
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Thu Feb 25 20:08:39 2021 +0900

    add local filter

commit ad900483cda9d7e3683c878dadd7517b259e943b
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Thu Feb 25 15:59:00 2021 +0900

    fix remove function, add comment

commit 3c2946986b814ef68fce656657c427b6c12df4e2
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Thu Feb 25 15:58:01 2021 +0900

    fix date -> datetime

commit 879fdf858e2f6fa0afe7ce5b3b7bfc48326d754a
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Wed Feb 24 19:31:18 2021 +0900

    add comment

commit 1631b1c6e8d6dbd48c61ef3c2706fcfa51a17294
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Wed Feb 24 16:46:58 2021 +0900

    change val name

commit 9225fa2689cd50dcc8dc15bd11c8c6731716e2bf
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Wed Feb 24 14:27:43 2021 +0900

    remove oldcode, comment

commit b1cd4f4fb0f81eb075ba5f83308c2b4a9ab3e4fa
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Wed Feb 24 14:27:13 2021 +0900

    fix const name

commit 19ae4fefd2fe346456e18c6eb47e884ee52878e5
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Wed Feb 24 14:20:23 2021 +0900

    remove unused function

commit 0f36e2f06d06557be6a0d159f5b2fa40676b8d21
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Wed Feb 24 14:15:59 2021 +0900

    fix val name

commit 1c2125ad323937570eee7e0e342600809d308ca7
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Wed Feb 24 14:12:07 2021 +0900

    fix dateformat

commit d38e8ffff24cdd2d1dbddbd9ac054d21b9944dd4
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Wed Feb 24 14:02:14 2021 +0900

    remove old code

commit 62d196b237b1afb37202b76f0d51e3cc091198ce
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Wed Feb 24 14:00:48 2021 +0900

    json generate: using new functions

commit 1575e5c54f50859862d3bcbe012d00f9e8ff9c69
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Wed Feb 24 14:00:00 2021 +0900

    fix dict nest

commit 2f012da8a7a45fbf28cd06981ab74a303ea56081
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Wed Feb 24 09:43:45 2021 +0900

    wip: fix

commit 695f8c37bdbdb75986b819b3f7e73a3b2571e80d
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Tue Feb 23 16:32:35 2021 +0900

    wip: add testcode fix jsontemplate, gen process

commit 9f668c112c5367a75876118b2f8dfc6f45c8203a
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Tue Feb 23 07:58:49 2021 +0900

    fix comment

commit 83da87e21434a5a0b523127b2cbe5d5149e5051a
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Tue Feb 23 07:40:49 2021 +0900

    update, add tests

commit 88ebf31d676b1935f527e38a6643082190160978
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Tue Feb 23 07:40:16 2021 +0900

    fix: sum function syntax error

commit 75aede21324d763ccc50bb1916ccb5ec624bbcd1
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Tue Feb 23 06:46:50 2021 +0900

    fix temporary func name

commit 996dff5d472de8fe43566fa0541d2a644a4ffd1f
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Tue Feb 23 06:45:54 2021 +0900

    fix flake8 error

commit 246529a470faf3baf8667b400345de737c316dbb
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Fri Feb 19 21:43:40 2021 +0900

    add test case, data

commit 6445f7346b277522776b8019c846237ace1308b2
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Fri Feb 19 21:43:26 2021 +0900

    fix function name

commit 8ad9f0f31197b2b2c821eded4f01c2062a051d76
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Fri Feb 19 12:38:01 2021 +0900

    wip: リファクタリング用のコメント追加, etc...

commit 8f1b81e18c5eb044c79661913ce465576f2c8901
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Fri Feb 19 12:26:07 2021 +0900

    wip: add comment

commit e11e73a3c018e11e4471e41a3a60bee2c97cd397
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Fri Feb 19 11:32:56 2021 +0900

    wip

commit 3e284245bdca86ffed654a411ee376eb43f4d3a4
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Fri Feb 19 10:15:38 2021 +0900

    wip fix function name, refactoring comment

commit 8ed623fb6cb7390c159a7ba35d1190495977751a
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Fri Feb 19 09:49:44 2021 +0900

    check dataset exist

commit 33932d1f85af6bee302afc516e7d6fe708b50642
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Fri Feb 19 09:41:16 2021 +0900

    wip: refactoring functions

commit ad5baff29cb9d1f52756b121a35237b1ce2274f7
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Fri Feb 19 08:02:34 2021 +0900

    wip: add function, comment

commit af5f68aad262d26414cb6dbbbd6650b9ac5d764d
Author: Hiroshi Sano <hrs.sano645@gmail.com>
Date:   Thu Feb 18 20:55:11 2021 +0900

    wip:split parse function

hrsano645 committed:
fix function name

hrsano645 committed:
add docstring

hrsano645 committed:
fix duplex line

hrsano645 committed:
fix comment

aktnk commented on patients.py:
```diff
@@ -171,10 +60,10 @@ def replace_nendai_format(src: str):
 
 
-def validate_opendata_dateformat(opendata_date_str):
+def validate_opendata_dateformat(opendata_date_str: str) -> datetime:
```
機能的には検証する（validate）というよりは、文字列からdatetimeオブジェクトに変換している感じですね。
変換できなかった理由は、規定外の日付フォーマットだったわけですが。

例えば、create_datetimeなどの方が良いのかもしれません。

hrsano645 replied:
そうですね。実際の処理はそうなってるので、生成は生成、検証は検証で分けてみます。

hrsano645 replied:
日付のバリデートについての必要性は、念のために必要だと考えていまして、どのタイミングでやろうか悩んでました。 -> https://github.com/hrsano645/covid19-gen-datajson-shizuokapref/issues/45#issuecomment-783746453

バリデーションしたときにエラーがあった場合は使えないデータとしてみるべきと考えていて、その流れで考えるとオープンデータを読み込むときにやるべきだと考えています。
（それで問題があったときにSlackに通知できたらいいなーと思ってますｗ

hrsano645 replied:
もともと関数名へのコメントでしたが、処理の中身含めて整理しなおしました。

-> https://github.com/hrsano645/covid19-gen-datajson-shizuokapref/commit/7273e0581305506d83a7bdfb34f0f38dc8e87b17

- 日付のバリデート関数(isvalid_opendata_dateformat)とバリデート関数を各項目に実行する関数(validate_dataset関数）を追加しました。
- main関数でバリデーションの処理を追加して、バリデートしてエラーがある文字列をデータ名とセットに出力して例外終了するようにしました。
- data.jsonの各データの処理で入れてあったバリデート処理を外しました。

aktnk commented on patients.py:
```diff
@@ -188,20 +77,41 @@ def validate_opendata_dateformat(opendata_date_str):
     return [
         start_datetime + timedelta(days=n)
-        for n in range((end_datetime - start_datetime).days)
+        for n in range((end_datetime - start_datetime).days + 1)
```
元の実装で”+1”がありませんが、動作確認結果はOKとのことでした。
・元の実装には実はバグがあったということでしょうか？

もしかすると、https://github.com/hrsano645/covid19-gen-datajson-shizuokapref/issues/37
との関連があると思います。
data.jsonを生成したタイミングで、
１）patients.csvの最終データが同じ日付のとき、
２）patients.csvの最終データが以前の日付のとき、
で動作を確認すると良いと思います。

hrsano645 replied:
元の実装はendからstartの日付を引いているので、実質1日足らない状態でした。（1/31-1/1は30になってしまう話です）そのため+1しています。

それ自体は問題になりますが、見えてこなかった理由は、最近の要請患者数が連日件数が上がる為です。

patients_summaryは、グラフとして用意したい全体の日付 （全体）と、patients.csvから陽性者の日ごとの出現数（patients.csv）の生成をしています。

- 全体: 2020/01/22 ~ end_datetime（現在はpatients.csvの終端行の日付。初期の実装はデータ生成日の次の日でした）
- patients.csv: 最初行の日付 ~ 終端行の日付

この処理部分は全体の日付の生成です。+1していない場合は冒頭に書いた通り1日少なくなってしまいます。終端行の日付-1の日付しか生成しないため、patients.csvの終端行日付から1日少なくずれてしまいます。

...といった事情で日付のラベルを作る際にこの処理にしています。ただ取り違えしやすいと思うので、書き方を工夫したかったのですが思いつかなかったです。

---

> data.jsonを生成したタイミングで、
> １）patients.csvの最終データが同じ日付のとき、
> ２）patients.csvの最終データが以前の日付のとき、
> で動作を確認すると良いと思います。

こちらはテスト入れたほうがいいと思うので追加しておきます。

hrsano645 replied:
こちらの変更は、+1がなぜ必要かのコメントで、もともとの実装だと想定した日付の日数が作れていないことがわかりました。

そのため（綺麗ではないのですが）+1を入れて数を合うようにしています。今後良い実装への変更も考えてテストを追加して日数計算が正しいかを検証できるようにしています。

> * `gen_datelist`関数にテストを追加する
>   
>   * [#46 (comment)](https://github.com/hrsano645/covid19-gen-datajson-shizuokapref/pull/46#discussion_r584242849)

-> https://github.com/hrsano645/covid19-gen-datajson-shizuokapref/commit/7fc30970969df0d84065b5aa381a7dc0f1cdeefe

- gen_datelist関数にテストを追加しました。
- テストはstartとendの範囲とその範囲の日数を見ています。

aktnk replied:
- 現時点のオープンデータでdata.jsonを生成してみました。
> 			{
> 				"日付": "2021-03-12T08:00:00.000Z",
> 				"小計": 17
> 			}
> 		]
既に、patients.csvには3/12のデータが掲載されており、正常に生成されていました。
- そこで故意にpatients.csvの3/12のデータを削除し、data.jsonを生成してみました。
> 			{
> 				"日付": "2021-03-11T08:00:00.000Z",
> 				"小計": 25
> 			}
> 		]
データがないのに、0で3/12のデータが生成されることはありませんでした。
⇒　問題ありません。

aktnk commented on patients.py:
```diff
@@ -188,20 +77,41 @@ def validate_opendata_dateformat(opendata_date_str):
 
 
-def gen_main_summary_data(details_of_confirmed_cases_filename: str):
+def change_datetimestr_zeropadding(target_datetime: datetime) -> str:
```
関数名からは、ゼロ埋め（zeroppaing)するようにも読み取れるので、
remove_leading_zeroやsuppress_zero、change_zero_suppressin_formatなどに変更した方がよいと思いました。

hrsano645 replied:
そうですね。changeだと除去ではないのでわかりづらいのはその通りだと思います。suppress_zeroがよさそうですね。

hrsano645 replied:
関数名の変更へのコメントを反映しました。

-> https://github.com/hrsano645/covid19-gen-datajson-shizuokapref/commit/c78abb8122d8d2bcea1ee631e9e4d4d27a0bde9f

aktnk commented on patients.py:
```diff
@@ -188,20 +77,41 @@ def validate_opendata_dateformat(opendata_date_str):
+    )
+
+
+def gen_jsonformat_datetime(t_datetime: datetime) -> str:
```
機能的には、jsonフォーマットに変換するよ言うよりも、規定時刻（UTCの08:00:00.000）に合わせている感じですね。

元々08:00:00.000Zなのですが、UTCなので、ローカルタイムだと、8:00+9:00＝17:00にしているのはなぜでしょう？

hrsano645 replied:
`08:00:00.000Z ` 添付の理由ですが、メトロ側がそうされているからと思われます。今のメトロ版 data.jsonも同様の表現です。

> 元々08:00:00.000Zなのですが、UTCなので、ローカルタイムだと、8:00+9:00＝17:00にしているのはなぜでしょう？

メトロ版のオープンデータ公開の時刻に合わせているからかもしれません。（確か会見が17時からだったと思います）

ですが、patientsのオープンデータAPI側を見てみると、公表日のフォーマット自体は同様ですが、時刻は`00:00:00.000Z`ですね。対策サイト側はこの部分を見ていないかもしれません。
https://portal.data.metro.tokyo.lg.jp/opendata-api/

hrsano645 replied:
この関数の意図としては、pythonのdatetimeをjsonに書き出すときの変換としていたので、中身含めてそのままで行こうと思っています。

aktnk commented:
差分では確認しにくかったので、動作フローを作成して関数名と動作の整合性を中心に確認してみました。
下記に作成した動作フロー図を掲載します。
- 全体動作フロー
![all](https://user-images.githubusercontent.com/13390370/109410080-fd092b00-79da-11eb-9d0a-efc47c16b3fd.png)
- gen_querents()の動作フロー
![gen_querents](https://user-images.githubusercontent.com/13390370/109410091-16aa7280-79db-11eb-9dff-cf89eab5de00.png)
- gen_patient()の動作フロー
![gen_patient](https://user-images.githubusercontent.com/13390370/109410100-29bd4280-79db-11eb-9834-fed7adf141df.png)
- gen_patient_summary()の動作フロー
![gen_patient_summary](https://user-images.githubusercontent.com/13390370/109410114-4f4a4c00-79db-11eb-8d77-addf0e21eac1.png)
- gen_inspections_summary()の動作フロー
![gen_inspections_summary](https://user-images.githubusercontent.com/13390370/109410122-62f5b280-79db-11eb-91db-0fe72ba69e7c.png)
- gen_main_summary()の動作フロー
![gen_main_summary](https://user-images.githubusercontent.com/13390370/109410129-73a62880-79db-11eb-94c2-22b339f3a762.png)

aktnk reviewed (changes requested):
実装内容を確認しました。
いくつか気になった点をコメントしました。
コメントをご確認いただき、最終的に変更するかどうかの判断はお任せします。

hrsano645 commented:
@aktnk 確認ありがとうございます。（フロー図すごいです。これって何か自動で生成させていますか？）
もう少し、処理の中身を手入れしたほうがいいところもあるのですが（例えばgen_関数でNoneを返すのに考慮してない 、関数でカプセル化した後での変数名の整理）、ひとまず目的は達成できたと思ってます。レビューの内容を修正したうえでmergeしようと思います。

修正については以下を行います。

- （日付のバリデーション処理と日付の生成を分ける -> ） **バリデーションするタイミングをオープンデータ生成時にして、問題があったときにエラーとして落とすようにする**
  - https://github.com/hrsano645/covid19-gen-datajson-shizuokapref/pull/46#discussion_r584242840
- `gen_datelist`関数にテストを追加する
  - https://github.com/hrsano645/covid19-gen-datajson-shizuokapref/pull/46#discussion_r584242849
- `change_datetimestr_zeropadding`関数の名前を `suppress_zero` に変更
  - https://github.com/hrsano645/covid19-gen-datajson-shizuokapref/pull/46#discussion_r584242864

aktnk commented:
> （フロー図すごいです。これって何か自動で生成させていますか？）

よく言えば、途中から自動生成です。
実装コードを読んで、PlantUML用のコードに自分が書いて、フロー図はPlantUMLに生成させています。
より詳しく説明すると、
１）PlantUMLサーバはDockerで立ち上げて、
２）VSCode＋PlantUML Pluginを使いPlantUMLサーバと連携させ、
３）自分が書いたUMLのアクティビティ図のコードをプレビューしたり、PNGファイルを生成させています。

hrsano645 committed:
不必要な変数代入を除去

hrsano645 committed:
change function name
https://github.com/hrsano645/covid19-gen-datajson-shizuokapref/pull/46#issuecomment-787451470

hrsano645 committed:
add gen_datelist test

hrsano645 committed:
バリデーションの処理を分離。バリデート関数にテスト追加

hrsano645 committed:
コメントと必要がないコードの除去

hrsano645 committed:
変数の整理

hrsano645 commented:
@aktnk の指摘を反映しました。変更のコミットとコメントを残しておきます。

> * （日付のバリデーション処理と日付の生成を分ける -> ） **バリデーションするタイミングをオープンデータ生成時にして、問題があったときにエラーとして落とすようにする**
>   
>   * [#46 (comment)](https://github.com/hrsano645/covid19-gen-datajson-shizuokapref/pull/46#discussion_r584242840)

-> https://github.com/hrsano645/covid19-gen-datajson-shizuokapref/commit/7273e0581305506d83a7bdfb34f0f38dc8e87b17

- 日付のバリデート関数(isvalid_opendata_dateformat)とバリデート関数を各項目に実行する関数(validate_dataset関数）を追加しました。
- main関数でバリデーションの処理を追加して、バリデートしてエラーがある文字列をデータ名とセットに出力して例外終了するようにしました。
- data.jsonの各データの処理で入れてあったバリデート処理を外しました。

> * `gen_datelist`関数にテストを追加する
>   
>   * [#46 (comment)](https://github.com/hrsano645/covid19-gen-datajson-shizuokapref/pull/46#discussion_r584242849)

-> https://github.com/hrsano645/covid19-gen-datajson-shizuokapref/commit/7fc30970969df0d84065b5aa381a7dc0f1cdeefe

- gen_datelist関数にテストを追加しました。
- テストはstartとendの範囲とその範囲の日数を見ています。

> * `change_datetimestr_zeropadding`関数の名前を `suppress_zero` に変更
>   
>   * [#46 (comment)](https://github.com/hrsano645/covid19-gen-datajson-shizuokapref/pull/46#discussion_r584242864)

-> https://github.com/hrsano645/covid19-gen-datajson-shizuokapref/commit/c78abb8122d8d2bcea1ee631e9e4d4d27a0bde9f

hrsano645 commented:
レビューコメントにも返信で変更内容を追加しました。

hrsano645 commented:
バリデーションがどのように機能するか説明します。

テストデータを用意して実行した様子です。二つのファイルを一部の行だけ残して、日付をエラーが起こるようにわざと変更しています。

test_number.csv

```csv
実施_年月日,全国地方公共団体コード,都道府県名,"検査実施_人数
（保健所）","検査実施_人数
（医療機関等）",備考
～2020/4/26,220001,静岡県,"2,637",828,累計用
2020/4/27,220001,静岡県,39,63,
/2020/4/28,220001,静岡県,83,73,
2020/4/29,220001,静岡県,63,38,
~2020/4/30,220001,静岡県,38,32,
220/5/1,220001,静岡県,97,57,

```

call_center.csv
```csv
受付_年月日,全国地方公共団体コード,都道府県名,市区町村名,相談件数
2020/2/10,220001,静岡県,,0
/2020/2/11,220001,静岡県,,0
2020/2/12,220001,静岡県,,47
2020/2/13,220001,静岡県,,49
~2020/2/14,220001,静岡県,,74
2020/2/15,220001,静岡県,,61
2020/2/16,220001,静岡県,,28
220/2/17,220001,静岡県,,164
2020/2/18,220001,静岡県,,199

```


定義された日付のフォーマットではない場合は以下のように表示されます。

patients.pyの実行結果

```
Traceback (most recent call last):
  File "patients.py", line 621, in <module>
    main()
  File "patients.py", line 553, in main
    raise DateValidateError(error_msg)
__main__.DateValidateError: バリデーションの結果、処理出来ない行があります。処理を終了します。:
call_center: /2020/2/11
call_center: ~2020/2/14
call_center: 220/2/17
inspections_summary: /2020/4/28
inspections_summary: ~2020/4/30
inspections_summary: 220/5/1
```

aktnk reviewed (approved):
@hrsano645 お疲れ様でした。
適切に修正されていることを確認しました。
また、data.jsonが問題なく生成されることも確認しました。

（参考）
patients.pyですが、各csvに対して似たような処理が繰り返される構造なので、
クラス化することで、凝集度を高めかつ、結合度を下げることができるかもしれないと思いました。

hrsano645 commented:
@aktnk ありがとうございます！mergeしますね。

> （参考）
> patients.pyですが、各csvに対して似たような処理が繰り返される構造なので、
> クラス化することで、凝集度を高めかつ、結合度を下げることができるかもしれないと思いました。

クラス化も考えてはいましたが、まずは現状の1ファイルのみでコードの状況整理をしたかったので、いきなりやらないでおきました。

このスクリプトもスクリプトといえる行数でもないので、モジュール分離をして各市町対応しやすい仕組みにできたらいいと思います。
